infra/event-sync/src/main/java/com/simprints/infra/eventsync/event/remote/EventRemoteDataSource.kt

```diff
@@ -51,15 +52,25 @@ internal class EventRemoteDataSource @Inject constructor(
 
     suspend fun getEvents(
         query: ApiRemoteEventQuery,
-        scope: CoroutineScope
-    ): ReceiveChannel<EnrolmentRecordEvent> {
+        scope: CoroutineScope,
+    ): EventDownSyncResult {
         return try {
-            val streaming = takeStreaming(query)
+            val response = takeStreaming(query)
+
+            val totalCount = response.headers()[COUNT_HEADER]?.toIntOrNull() ?: 0
+            val isTotalLowerBound = response
+                .headers()[IS_COUNT_HEADER_LOWER_BOUND]
+                ?.toBoolean() ?: false // If not present, assume it's not lower bound
+
+            val streaming = response.body()?.byteStream() ?: ByteArrayInputStream(byteArrayOf())
             Simber.tag("SYNC").d("[EVENT_REMOTE_SOURCE] Stream taken")
 
-            scope.produce(capacity = CHANNEL_CAPACITY_FOR_PROPAGATION) {
-                parseStreamAndEmitEvents(streaming, this)
-            }
+            EventDownSyncResult(
+                totalCount = totalCount.takeUnless { isTotalLowerBound },
```

Comment:
Why not take `totalCount` if it was sent but is lower bound?

Reply:
It messes up the UI slightly. The problem is that if there are more items to down sync header returns (there are technical reasons for it on BE), we cannot meaningfully show the percent indicator and show a continuous one instead. That is the same not having the count in the first place, so why bother passing it around :D